0.2.0-rc1 - Address PR comments for #37

valydia opened the pull request:
@marius-m made a great job on #37  - just addressing the final comments...

valydia commented:
Hey folks! Is there any chance this PR could be reviewed / merged? I'd be happy to address any feedback on it 👍 ... cc @ben-j69

ben-j69 commented:
@valydia yes it is planned but we are very busy for the moment.

ben-j69 commented on src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:
```diff
@@ -153,12 +205,13 @@ open class HiddenSecretsPlugin : Plugin<Project> {
+         * @param clean whether to overwrite existing files
          */
-        fun copyCppFiles() {
+        fun copyCppFiles(clean: Boolean) {
```
A parameter called `overwrite` would be clearer. And to not change too much the code and the actual behaviour a default value should be set. So the actual calls won't change
`        fun copyCppFiles(overwrite: Boolean = false) {}`

ben-j69 commented on src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:
```diff
@@ -168,23 +221,92 @@ open class HiddenSecretsPlugin : Plugin<Project> {
-            getKotlinFile()?.let {
-                println("$KOTLIN_FILE_NAME already exists")
-                return
+        fun copyKotlinFile(clean: Boolean) {
```
`Clean` is unclear here too.

ben-j69 commented on src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:
```diff
@@ -93,6 +100,42 @@ open class HiddenSecretsPlugin : Plugin<Project> {
+        }
+
+        /**
+         * Get properties file to hide secrets from
```
Get properties from the provided file

ben-j69 commented on src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:
```diff
@@ -236,69 +367,56 @@ open class HiddenSecretsPlugin : Plugin<Project> {
-                    kotlinPackage = packageName
-                }
+                hideSecret(
+                    keyName = keyName,
```
Not need to repeat names and vars: 
`hideSecret(keyName, packageName, obfuscatedKey)`

ben-j69 commented on src/main/kotlin/com/klaxit/hiddensecrets/HiddenSecretsPlugin.kt:
```diff
@@ -236,69 +367,56 @@ open class HiddenSecretsPlugin : Plugin<Project> {
+                props.entries.forEach { entry ->
+                    val keyName = entry.key as String
+                    val obfuscatedKey = Utils.encodeSecret(entry.value as String, packageName)
+                    hideSecret(
```
Not need to repeat names and vars: 
`hideSecret(keyName, packageName, obfuscatedKey)`

ben-j69 reviewed (changes requested):
@valydia, thanks for the work. Can you rebase and adjust some last comments.

I have time allowed to close this improvement.

It will be merge and release in 0.2.0-rc1

valydia commented:
Thanks for the review @ben-j69 ! I've addressed your comment in https://github.com/klaxit/hidden-secrets-gradle-plugin/pull/47/commits/a3cbe0462e136433f037b8295ef7c99541938bb7

ben-j69 commented:
Thanks @valydia , I will merge it in 0.2.0-rc1, test and adjust if necessary and then create a new release ;)

ben-j69 commented:
Hello @valydia I made a mistake on rebase and I hade to manually copy the code to make a new commit here https://github.com/klaxit/hidden-secrets-gradle-plugin/pull/56 is it ok for you ? Not sure your name and @marius-m will appear on the repo.

If not can you create a new PR to master and I will recreate a 0.2.0

Thanks

valydia commented:
Hi @ben-j69 ! Thanks for letting me know! You can proceed with your new commit! Cheers
